akka-sample-sharding-java/killrweather/src/main/java/sample/killrweather/WeatherStation.java

```diff
@@ -0,0 +1,241 @@
+package sample.killrweather;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.PostStop;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+import akka.cluster.sharding.typed.javadsl.ClusterSharding;
+import akka.cluster.sharding.typed.javadsl.Entity;
+import akka.cluster.sharding.typed.javadsl.EntityTypeKey;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Generally geo-based data, unless it is a floating station.
+ *
+ * This is just an example: cluster sharding would be overkill for just keeping a small amount of data,
+ * but becomes useful when you have a collection of 'heavy' actors (in terms of processing or state)
+ * so you need to distribute them across several nodes.
+ *
+ * A sharded `WeatherStation` has a set of recorded datapoints
+ * For each weather station common cumulative computations can be run:
+ * aggregate, averages, high/low, topK (e.g. the top N highest temperatures).
+ *
+ * Note that since this station is not storing its state anywhere else than in JVM memory, if Akka Cluster Sharding
+ * rebalances it - moves it to another node because of cluster nodes added removed etc - it will lose all its state.
+ * For a sharded entity to have state that survives being stopped and started again it needs to be persistent,
+ * for example by being an EventSourcedBehavior.
+ */
+final class WeatherStation extends AbstractBehavior<WeatherStation.Command> {
+
+  // setup for using WeatherStations through Akka Cluster Sharding
+  // these could also live elsewhere and the WeatherStation class be completely
+  // oblivious to being used in sharding
+  public static final EntityTypeKey<Command> TypeKey =
+    EntityTypeKey.create(WeatherStation.Command.class, "WeatherStation");
+
+  public static void initSharding(ActorSystem<?> system) {
```

Comment:
Maybe point this out in the readme, and that this is a common pattern for sharded actors